### PR TITLE
Upgrade to Jint 3.1.0

### PR DESCRIPTION
**Pull request by lahma**
# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## Description

* upgrade Jint to Jint 3.1, remove obsolete member usage with better ones
* small cleanup to API usage
* add missing license expression to csproj
* treat warnings as errors to catch obsolete members

### CODE REVIEW

File: src/AngleSharp.Js/AngleSharp.Js.csproj [hunk 1]
```diff
@@ -1,28 +1,24 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <AssemblyName>AngleSharp.Js</AssemblyName>
-    <RootNamespace>AngleSharp.Js</RootNamespace>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard2.0;net6.0;net7.0;net8.0</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netstandard2.0;net462;net472;net6.0;net7.0;net8.0</TargetFrameworks>
-    <SignAssembly>true</SignAssembly>
```
Comment (FlorianRappl) [inline src/AngleSharp.Js/AngleSharp.Js.csproj:7]: This is a breaking change - can you please leave such things untouched? Thanks!

File: src/Directory.Build.props [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <Version>1.0.0</Version>
     <LangVersion>latest</LangVersion>
     <SignAssembly>true</SignAssembly>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (FlorianRappl) [inline src/Directory.Build.props:8]: Wouldn't go so far; a warning is a warning.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]

File: src/AngleSharp.Js/AngleSharp.Js.csproj (hunk 1 above)
Comment (lahma) [reply src/AngleSharp.Js/AngleSharp.Js.csproj:7]: Which part? 

* Assembly name and namespace default to project's name (`AngleSharp.Js`) 
* Directory.build props gives the `<SignAssembly>true</SignAssembly>` and `<SignAssembly>true</SignAssembly>`
* targets are the same as before, defaults for all platforms and on windows the additional full framework ones

File: src/Directory.Build.props (hunk 2 above)
Comment (lahma) [reply src/Directory.Build.props:8]: I can revert if you wish, you just have the exact same configuration for main `AngleSharp` project:

https://github.com/AngleSharp/AngleSharp/blob/devel/src/Directory.Build.props#L8

File: src/AngleSharp.Js/AngleSharp.Js.csproj (hunk 1 above)
Comment (FlorianRappl) [reply src/AngleSharp.Js/AngleSharp.Js.csproj:7]: (comment was about SignAssembly)

Yeah did not see it - only saw the removal. True its already part of the build props and thus should not be repeated. Thanks for clarifying.

File: src/Directory.Build.props (hunk 2 above)
Comment (FlorianRappl) [reply src/Directory.Build.props:8]: Yes its a good point, for consistency it makes sense. Let's keep it!

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**